File: SWIG/Examples/test-suite/ruby/abstract_signature_runme.rb
require 'abstract_signature'

include Abstract_signature

#
# Shouldn't be able to instantiate Abstract_foo, because it declares
# a pure virtual function.
#

exceptionRaised = false
begin
  Abstract_foo.new
rescue NameError
  exceptionRaised = true
rescue TypeError
  # In Ruby 1.8 the exception raised is:
  # TypeError: allocator undefined for Abstract_signature::Abstract_foo
	exceptionRaised = true
ensure
  raise RuntimeError unless exceptionRaised
end

#
# Shouldn't be able to instantiate an Abstract_bar either, because it doesn't
# implement the pure virtual function with the correct signature.
#

exceptionRaised = false
begin
  Abstract_bar.new
rescue NameError
  exceptionRaised = true
rescue TypeError
  # In Ruby 1.8 the exception raised is:
  # TypeError: allocator undefined for Abstract_signature::Abstract_bar
	exceptionRaised = true
ensure
  raise RuntimeError unless exceptionRaised
end

File: Examples/test-suite/lua/abstract_signature_runme.lua
local v=require("abstract_signature")

--[[ Shouldn't be able to instantiate Abstract_foo, because it declares
     a pure virtual function. ]]
assert(not pcall(function ()
    local foo = v.abstract_foo()
    foo:meth(1)
end))


--[[ Shouldn't be able to instantiate an Abstract_bar either, because it doesn't
     implement the pure virtual function with the correct signature. ]]
assert(not pcall(function ()
    local foo = v.abstract_bar()
    foo:meth(1)
end))
